### PR TITLE
Fixing CMake build on Windows for MSVC and Clion, addressing warnings.

### PR DESCRIPTION
**Pull request by a-day-old-bagel**
This does not completely fix the issues, and it may well cause other issues, as I've only tested these changes running CMake with MSVC in Windows (both in CLion and VS).

Please let me know if we need to keep the /WX flag for MSVC builds under CMake, since that's causing most of the problems (warnings treated as errors causes code inside GLM and GLAD to not compile)

### CODE REVIEW

File: src/client/network/client.cpp [hunk 1]
```diff
@@ -53,16 +53,16 @@ void Client::sendChunkRequest(const ChunkPosition &position)
                             ENET_PACKET_FLAG_RELIABLE);
 }
 
-void Client::onPeerConnect(ENetPeer &peer)
+void Client::onPeerConnect(ENetPeer &peer) // FIXME: unreferenced formal parameter warning in MSVC
 {
 }
 
-void Client::onPeerDisconnect(ENetPeer &peer)
+void Client::onPeerDisconnect(ENetPeer &peer) // FIXME: unreferenced formal parameter warning in MSVC
```
Comment (Hopson97) [inline src/client/network/client.cpp]:
```suggestion
void Client::onPeerDisconnect([[maybe_unused]] ENetPeer &peer) // FIXME: unreferenced formal parameter warning in MSVC
```

File: src/client/network/client.cpp [hunk 2]
```diff
@@ -53,16 +53,16 @@ void Client::sendChunkRequest(const ChunkPosition &position)
                             ENET_PACKET_FLAG_RELIABLE);
 }
 
-void Client::onPeerConnect(ENetPeer &peer)
+void Client::onPeerConnect(ENetPeer &peer) // FIXME: unreferenced formal parameter warning in MSVC
```
Comment (Hopson97) [inline src/client/network/client.cpp]: For unused variables, this should be able to help with this

```suggestion
void Client::onPeerConnect([[maybe_unused]] ENetPeer &peer) 
```

https://en.cppreference.com/w/cpp/language/attributes/maybe_unused

File: CMakeLists.txt [hunk 3]
```diff
@@ -21,7 +22,7 @@ function (set_flags target)
         target_compile_options(${target} 
             PRIVATE 
             /W4 
-            /WX
+#            /WX
```
Comment (Hopson97) [inline CMakeLists.txt]: It's probably best we just get rid of this option entirely, not sure if anyone really likes warning being treated as errors 
```suggestion
```

**Comment by a-day-old-bagel** [comment]
Aw geez. Apparently Ubuntu doesn't like CMake trying to link to glm as a header-only interface.

**Comment by Hopson97** [comment]
I think it is possible to check the platform in the CMakeLists, you could make the "breaking" section windows only

**Comment by Hopson97** [comment]
Also this seems to be your fix, would it be possible to add these into this PR as well? Thanks!

https://github.com/Hopson97/open-builder/issues/30#issuecomment-568453507

Also with windows only code in cmake (to fix these unit tests) you can do 

```cmake
if(WIN32)
   # Put the windows code/ glm fixes here
endif()
```

https://stackoverflow.com/questions/9160335/os-specific-instructions-in-cmake-how-to

**Comment by a-day-old-bagel** [comment]
Working on it, sorry for the delay - my Windows machine's GPU blew up so I'm trying to work around that in addition to the holidays having just happened.

**Comment by a-day-old-bagel** [comment]
As far as adding the GLM config modules into the repo, would it be acceptable to add a secion header "Windows [CMake]" to the readme under the existing "Windows [Visual Studio]" section, where we instruct the user to copy and paste the two modules into their GLM directory?
And then would we include the modules themselves in the `cmake_modules` directory?

**Comment by Hopson97** [comment]
No worries, its a busy time of year :)

I think it might be worth making a new section for "Window [CMake/ CLion], as the these shouldn't effect the current "hacky" visual studio set up. 

And yeah I guess that would be a good place to put them, but I am not sure if that would effect the current build as CMake will look in that folder for the FindSFML.cmake so it might look at those cmake files too, but not sure.

Btw, how do you feel about this solution to the issue: #36

They add GLM as a submodule, meaning it should "just work", but I am not sure as I haven't really used these before.

**Comment by a-day-old-bagel** [comment]
Including glm as a submodule is what I have typically done in my own projects. I was exploring these other solutions in the spirit of not contesting design decisions you make, but rather changing as little as possible to produce a streamlined, functioning build process.
Including dependencies as submodules is a fairly good and portable solution for projects that are meant to compile in many environments. If you decide to go that route I think you'll be pleased with it.
Furthermore, if you do decide to move forward with that strategy, you might also consider doing the same with your other dependencies, instead of copy/pasting a minimal version of their sources directly into your own repo as you've done with enet, for example.

**Comment by Hopson97** [comment]
It's looking very attractive now. Just need to make sure I can specify the correct glm version for the submodule, and then probably will go down with that route.

**Comment by BrunoSilvaFreire** [comment]
Reading the conversation, since this is improving CLion support, probably it would be nice to include a couple team-shared IntelliJ files to improve even more experience when using CLion.
I've created a pull request [here](https://github.com/a-day-old-bagel/open-builder/pull/1).
This allows some cool stuff like mark specific directories as source, adding custom team-shared configurations, etc.
Here is the end result:

**Comment by Hopson97** [comment]
Solved by #87

**Pull request closed without merging**